tests/Unit/Domain/Test_FlatLogicalMetric.cpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/FlatLogicalMetric.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/LogicalCoordinates.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+template <size_t Dim>
+auto make_affine_coord_map() {
+  using AffineMap = domain::CoordinateMaps::Affine;
+  if constexpr (Dim == 1) {
+    return domain::CoordinateMap<Frame::ElementLogical, Frame::Inertial,
+                                 AffineMap>{{-1., 1., 0., M_PI}};
+  } else if constexpr (Dim == 2) {
+    using AffineMap2D =
+        domain::CoordinateMaps::ProductOf2Maps<AffineMap, AffineMap>;
+    return domain::CoordinateMap<Frame::ElementLogical, Frame::Inertial,
+                                 AffineMap2D>{
+        {{-1., 1., 0., M_PI}, {-1., 1., 0., M_PI}}};
+  } else {
+    using AffineMap3D =
+        domain::CoordinateMaps::ProductOf3Maps<AffineMap, AffineMap, AffineMap>;
+    return domain::CoordinateMap<Frame::ElementLogical, Frame::Inertial,
+                                 AffineMap3D>{
+        {{-1., 1., 0., M_PI}, {-1., 1., 0., M_PI}, {-1., 1., 0., M_PI}}};
+  }
+}
+
+template <size_t Dim>
+void test_flat_logical_metric() {
+  CAPTURE(Dim);
+  const Mesh<Dim> mesh{3, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  const size_t num_points = mesh.number_of_grid_points();
+  const auto logical_coords = logical_coordinates(mesh);
+  const auto coord_map = make_affine_coord_map<Dim>();
+  const auto inertial_coords = coord_map(logical_coords);
+  const auto jacobian = coord_map.jacobian(logical_coords);
+  const auto inv_jacobian = coord_map.inv_jacobian(logical_coords);
+  tnsr::ii<DataVector, Dim, Frame::ElementLogical> flat_logical_metric{
+      num_points};
+  domain::flat_logical_metric(make_not_null(&flat_logical_metric), jacobian);
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j < Dim; ++j) {
+      CHECK_ITERABLE_APPROX(
+          flat_logical_metric.get(i, j),
+          DataVector(num_points, i == j ? square(M_PI_2) : 0.));
```

Comment:
Test a case that's less trivial.  Preferably something where you don't have all the matrices diagonal, but if that's a pain at least make it non-isotropic.

Reply:
added another test in a fixup